docs/source/en/model_doc/bert.md

```diff
@@ -61,6 +61,30 @@ This model was contributed by [thomwolf](https://huggingface.co/thomwolf). The o
     
 - The model must predict the original sentence, but has a second objective: inputs are two sentences A and B (with a separation token in between). With probability 50%, the sentences are consecutive in the corpus, in the remaining 50% they are not related. The model has to predict if the sentences are consecutive or not.
 
+### Using Scaled Dot Product Attention (SDPA)
+
+PyTorch includes a native scaled dot-product attention operator (SDPA) as part of `torch.nn.functional`. This function 
+encompasses several implementations that can be applied depending on the inputs and the hardware in use. See the 
+[official documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) 
+or the [GPU Inference](https://huggingface.co/docs/transformers/main/en/perf_infer_gpu_one#pytorch-scaled-dot-product-attention)
+page for more information.
+
+SDPA is used by default for `torch>=2.1.1` when an implementation is available, but you may also set 
+`attn_implementation="sdpa"` in `from_pretrained()` to explicitly request SDPA to be used.
+
+```
+from transformers import BertModel
+
+model = BertModel.from_pretrained("bert-base-uncased", torch_dtype=torch.float16, attn_implementation="sdpa")
+...
+```
+
+For the best speedups, we recommend loading the model in half-precision (e.g. `torch.float16`).
+
+On a local benchmark (A100-80GB, CPUx12, RAM 96.6GB, OS Ubuntu 22.04), we have seen speedups as high as 40%+ for
+training and 20%+ for inference. The improvements are higher for larger batch sizes and longer sequence lengths.
```

Comment:
Do we have a graph of the running times across batch sizes? I think this would be most user for users to decide if they want to use this attention implementation

Reply:
I don't know how to generate that graph that showed up for flash attention (do you have a resource?), but I can add the latest benchmarks as a table if you think it helps. 

These are the latest ones - [training](https://gist.github.com/hackyon/2bfcfb59666fab23655f6e62bc0752ba) and [inference](https://gist.github.com/hackyon/c3d829d06144b98c6b092a66733fa1a9).

One caveat is that, with SDPA, the gains are quite dependent on hardware, torch version, dtype, and the selected kernel. Even if the user uses SDPA, the gains they see could vary a lot depending on their configuration.

Reply:
Yes please - a table would be great - it's far more useful for users to see many data points and how much each variable affects things

Reply:
Sure, added the benchmark tables to the docs.